pkg/sdkserver/sdkserver.go

```diff
@@ -985,41 +999,238 @@ func (s *SDKServer) GetList(ctx context.Context, in *alpha.GetListRequest) (*alp
+	}
+
+	name := in.List.Name
+	s.logger.WithField("name", name).Debug("Update List -- Currently only used for Updating Capacity")
+
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Pull in variable Max Capacity from CRD instead of hard-coded number here.
+	if in.List.Capacity < 0 || in.List.Capacity > 1000 {
+		return nil, errors.Errorf("OUT_OF_RANGE. Capacity must be within range [0,1000]. Found Capacity: %d", in.List.Capacity)
+	}
+
+	if _, ok := gs.Status.Lists[name]; ok {
+		batchList := s.gsListUpdates[name]
+		batchList.capacitySet = &in.List.Capacity
+		s.gsListUpdates[name] = batchList
+		// Queue up the Update for later batch processing by updateLists.
+		s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+		return &alpha.List{}, nil
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", name)
 }
 
-// AddListValue returns the updated List.
+// AddListValue collapses all append a value to the end of a List requests into a single UpdateList request.
+// Returns NOT_FOUND if the List does not exist.
+// Returns ALREADY_EXISTS if the value is already in the List.
+// Returns OUT_OF_RANGE if the List is already at Capacity.
 // [Stage:Alpha]
 // [FeatureFlag:CountsAndLists]
 func (s *SDKServer) AddListValue(ctx context.Context, in *alpha.AddListValueRequest) (*alpha.List, error) {
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement Me
-	return nil, errors.Errorf("Unimplemented -- AddListValue coming soon")
+	s.logger.WithField("name", in.Name).Debug("Add List Value")
+
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	if list, ok := gs.Status.Lists[in.Name]; ok {
+		batchList := s.gsListUpdates[in.Name]
+		// Verify room to add another value
+		var capacity int64
+		if batchList.capacitySet != nil {
+			capacity = *batchList.capacitySet
+		} else {
+			capacity = int64(len(list.Values) + len(batchList.listAppend) - len(batchList.listDelete))
+		}
+		if list.Capacity <= capacity {
+			return nil, errors.Errorf("OUT_OF_RANGE. No available capacity. Current Capacity: %d, List Size: %d", list.Capacity, len(list.Values))
+		}
+		// Verify value does not already exist in the list
+		// TODO: This does not check batched but not yet applied append / remove values. Should we do this?
+		// (Easy to check not yet applied values, hard to check removed and re-added values.) I'm
+		// thinking this would be better / easier to do as part of the batch apply update to list, and
+		// not verify here.
+		for _, val := range list.Values {
+			if in.Value == val {
+				return nil, errors.Errorf("ALREADY_EXISTS. Value: %s already in List: %s", in.Value, in.Name)
```

Comment:
So from the design for `ListAppend` (assuming we're still calling it that) [we have](https://github.com/googleforgames/agones/issues/2716#:~:text=Returns%20false%2C%20if%20the%20value%20already%20exists%20in%20the%20list%2C%20or%20if%20the%20list%20is%20already%20at%20capacity%20(to%20the%20latest%20knowledge%20of%20the%20SDK).):

> Returns false, if the value already exists in the list, or if the list is already at capacity (to the latest knowledge of the SDK).

(or at least on a best effort basis).

So we might need to modify the return type on this function?